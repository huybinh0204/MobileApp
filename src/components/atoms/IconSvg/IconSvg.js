import PropTypes from 'prop-types';
import React from 'react';
import BaseText from '_components/atoms/BaseText/BaseText';
import icons from './iconList';

const IconSvg = ({ icon, width, height, style, testID, ...rest }) => {
  if (!icons[icon]) {
    return <BaseText testID={testID}>not Found</BaseText>;
  }

  return icons[icon]({ style, width, height, testID, ...rest });
};

IconSvg.defaultProps = {
  width: 20,
  height: 20,
};

IconSvg.propTypes = {
  icon: PropTypes.string.isRequired,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default IconSvg;
