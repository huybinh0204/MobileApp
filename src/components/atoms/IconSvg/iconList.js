import React from 'react';
// Action icons
import BackIcon from '_assets/ActionIcons/back.svg';
import Close from '_assets/ActionIcons/close.svg';
import CloseAlert from '_assets/ActionIcons/closeAlert.svg';
import CloseError from '_assets/ActionIcons/closeError.svg';
import CloseInfo from '_assets/ActionIcons/closeInfo.svg';
import CloseSuccess from '_assets/ActionIcons/closeSuccess.svg';
import CloseWhite from '_assets/ActionIcons/closeWhite.svg';
import Share from '_assets/ActionIcons/share.svg';
import ToastAlert from '_assets/ActionIcons/toastAlert.svg';
import ToastInfo from '_assets/ActionIcons/toastInfo.svg';
// Banners
import DebitCardBanner from '_assets/Banners/debit_card_banner.svg';
import Playlist1 from '_assets/finEd/playlist-1.svg';
// Fined
import FinEd from '_assets/GeneralIcons/finEd.svg';
import FinEdInactive from '_assets/GeneralIcons/finEdInactive.svg';
// General Icons - Normal
import Accounts from '_assets/GeneralIcons/normal/accounts.svg';
import AccountsInactive from '_assets/GeneralIcons/normal/accountsInactive.svg';
import Bank from '_assets/GeneralIcons/normal/bank.svg';
import BankBlack from '_assets/GeneralIcons/normal/bankBlack.svg';
import BankInactive from '_assets/GeneralIcons/normal/bankInactive.svg';
import Card from '_assets/GeneralIcons/normal/card.svg';
import CardInactive from '_assets/GeneralIcons/normal/cardInactive.svg';
import EditIcon from '_assets/GeneralIcons/normal/edit.svg';
import GiftOutlined from '_assets/GeneralIcons/normal/giftOutlined.svg';
import HidePassword from '_assets/GeneralIcons/normal/hidePassword.svg';
import Home from '_assets/GeneralIcons/normal/home.svg';
import HomeInactive from '_assets/GeneralIcons/normal/homeInactive.svg';
import InstantIcon from '_assets/GeneralIcons/normal/instant.svg';
import ShowPassword from '_assets/GeneralIcons/normal/showPassword.svg';
// General Icons - Rounded background
import AboutSettings from '_assets/GeneralIcons/roundedBackground/about.svg';
import AccountInfoSettings from '_assets/GeneralIcons/roundedBackground/accountInfo.svg';
import ActivatedCard from '_assets/GeneralIcons/roundedBackground/activatedCard.svg';
import AddBankAccount from '_assets/GeneralIcons/roundedBackground/addBankAccount.svg';
import Atm from '_assets/GeneralIcons/roundedBackground/atm.svg';
import ChangePin from '_assets/GeneralIcons/roundedBackground/changePin.svg';
import ContactSetting from '_assets/GeneralIcons/roundedBackground/contact.svg';
import DebitCard from '_assets/GeneralIcons/roundedBackground/debitCard.svg';
import DirectDepositSettings from '_assets/GeneralIcons/roundedBackground/directDepositSettings.svg';
import EmailCircled from '_assets/GeneralIcons/roundedBackground/email.svg';
import FaqSettings from '_assets/GeneralIcons/roundedBackground/faq.svg';
import Finances from '_assets/GeneralIcons/roundedBackground/finances.svg';
import FinancesInactive from '_assets/GeneralIcons/roundedBackground/financesInactive.svg';
import FindATMOption from '_assets/GeneralIcons/roundedBackground/findAtmOption.svg';
import FindDepositCash from '_assets/GeneralIcons/roundedBackground/findDepositCash.svg';
import HomeCircled from '_assets/GeneralIcons/roundedBackground/home.svg';
import Key from '_assets/GeneralIcons/roundedBackground/key.svg';
import Lightning from '_assets/GeneralIcons/roundedBackground/lightning.svg';
import LinkCard from '_assets/GeneralIcons/roundedBackground/linkCard.svg';
import LockCard from '_assets/GeneralIcons/roundedBackground/lockCard.svg';
import OverdraftProtectionSettings from '_assets/GeneralIcons/roundedBackground/overdraftProtectionSettings.svg';
import PhoneCircled from '_assets/GeneralIcons/roundedBackground/phone.svg';
import ReplaceCard from '_assets/GeneralIcons/roundedBackground/replaceCard.svg';
import Statements from '_assets/GeneralIcons/roundedBackground/statements.svg';
import TermsSettings from '_assets/GeneralIcons/roundedBackground/terms.svg';
import UserCircled from '_assets/GeneralIcons/roundedBackground/user.svg';
import ShopDeals from '_assets/GeneralIcons/shopDeals.svg';
// General Icons - Special
import ActivateCard from '_assets/GeneralIcons/special/activateCard.svg';
import BankCircle from '_assets/GeneralIcons/special/bank.svg';
import DepositCash from '_assets/GeneralIcons/special/depositCash.svg';
import DirectDeposit from '_assets/GeneralIcons/special/directDeposit.svg';
import EarlyPaycheck from '_assets/GeneralIcons/special/earlyPaycheck.svg';
import OverdraftProtection from '_assets/GeneralIcons/special/overdraftProtection.svg';
import AngleLeft from '_assets/GeneralIcons/system/angleLeft.svg';
import AngleRight from '_assets/GeneralIcons/system/angleRight.svg';
// General Icons - System
import ArrowLeft from '_assets/GeneralIcons/system/arrowLeft.svg';
import ArrowRight from '_assets/GeneralIcons/system/arrowRight.svg';
import ArrowRightActive from '_assets/GeneralIcons/system/arrowRightActive.svg';
import ArrowRightInactive from '_assets/GeneralIcons/system/arrowRightInactive.svg';
import ArrowsDown from '_assets/GeneralIcons/system/arrowsDown.svg';
import BurgerMenu from '_assets/GeneralIcons/system/burgerMenu.svg';
import CheckmarkLines from '_assets/GeneralIcons/system/checkmarkLines.svg';
import CircledCheckmark from '_assets/GeneralIcons/system/circledCheckmark.svg';
import CircledExclamation from '_assets/GeneralIcons/system/circledExclamation.svg';
import Credit from '_assets/GeneralIcons/system/credit.svg';
import Debit from '_assets/GeneralIcons/system/debit.svg';
import GreenArrowDown from '_assets/GeneralIcons/system/greenArrowDown.svg';
import Menu from '_assets/GeneralIcons/system/menu.svg';
import Reload from '_assets/GeneralIcons/system/reload.svg';
// Illustrations
import AccountSummary from '_assets/illustrations/account-summary.svg';
import CardSuccessActivation from '_assets/illustrations/activate-card-success-illustration.svg';
import CardJumpingIllustration from '_assets/illustrations/card-jumping.svg';
import CheckEmail from '_assets/illustrations/check-email.svg';
import Contact from '_assets/illustrations/contact.svg';
import ErrorIllustration from '_assets/illustrations/error-illustration.svg';
import GetPaidEarlyIllustration from '_assets/illustrations/get-paid-early-illustration.svg';
import LegalAgreementIllustration from '_assets/illustrations/legal-agreement-illustration.svg';
import ParachuteCard from '_assets/illustrations/parachute-card-illustration.svg';
import SetDebitCardPin from '_assets/illustrations/set-debit-card-pin.svg';
import SuccessIllustration from '_assets/illustrations/success-illustration.svg';
// Logos
import Logo from '_assets/logos/logo.svg';
import LogoSpecial from '_assets/logos/logoSpecial.svg';
import LogoVertical from '_assets/logos/logoVertical.svg';
// Marketplace Icons
import Amplify from '_assets/MarketplaceIcons/amplify.svg';
import Better from '_assets/MarketplaceIcons/better.svg';
import Fiona from '_assets/MarketplaceIcons/fiona.svg';
import HavenLife from '_assets/MarketplaceIcons/havenLife.svg';
import IconSavingPlan from '_assets/MarketplaceIcons/iconSavingPlan.svg';
import Jerry from '_assets/MarketplaceIcons/jerry.svg';
import RocketLawyer from '_assets/MarketplaceIcons/rocketLawyer.svg';
import Savi from '_assets/MarketplaceIcons/savi.svg';
import Self from '_assets/MarketplaceIcons/self.svg';
import TrustAndWill from '_assets/MarketplaceIcons/trustandwill.svg';
import UnestAdvisers from '_assets/MarketplaceIcons/unestAdvisers.svg';
// Marketplace Logos
import AmplifyLogo from '_assets/MarketplaceLogos/amplify.svg';
import BetterLogo from '_assets/MarketplaceLogos/better.svg';
import FionaLogo from '_assets/MarketplaceLogos/fiona.svg';
import HavenLifeLogo from '_assets/MarketplaceLogos/haven_life.svg';
import IconSavingPlanLogo from '_assets/MarketplaceLogos/icon_saving_plan.svg';
import JerryLogo from '_assets/MarketplaceLogos/jerry.svg';
import RocketLawyerLogo from '_assets/MarketplaceLogos/rocket_lawyer.svg';
import SaviLogo from '_assets/MarketplaceLogos/savi.svg';
import SelfLogo from '_assets/MarketplaceLogos/self.svg';
import TrustAndWillLogo from '_assets/MarketplaceLogos/trust_will.svg';
import UnestAdvisersLogo from '_assets/MarketplaceLogos/unest.svg';

export default {
  amplifyLogo: (props) => <AmplifyLogo {...props} />,
  betterLogo: (props) => <BetterLogo {...props} />,
  fionaLogo: (props) => <FionaLogo {...props} />,
  havenLifeLogo: (props) => <HavenLifeLogo {...props} />,
  iconSavingPlanLogo: (props) => <IconSavingPlanLogo {...props} />,
  jerryLogo: (props) => <JerryLogo {...props} />,
  rocketLawyerLogo: (props) => <RocketLawyerLogo {...props} />,
  saviLogo: (props) => <SaviLogo {...props} />,
  selfLogo: (props) => <SelfLogo {...props} />,
  trustAndWillLogo: (props) => <TrustAndWillLogo {...props} />,
  unestAdvisersLogo: (props) => <UnestAdvisersLogo {...props} />,
  amplify: (props) => <Amplify {...props} />,
  better: (props) => <Better {...props} />,
  fiona: (props) => <Fiona {...props} />,
  havenLife: (props) => <HavenLife {...props} />,
  iconSavingPlan: (props) => <IconSavingPlan {...props} />,
  jerry: (props) => <Jerry {...props} />,
  rocketLawyer: (props) => <RocketLawyer {...props} />,
  savi: (props) => <Savi {...props} />,
  self: (props) => <Self {...props} />,
  trustandwill: (props) => <TrustAndWill {...props} />,
  unestAdvisers: (props) => <UnestAdvisers {...props} />,
  shopDeals: (props) => <ShopDeals {...props} />,
  logo: (props) => <Logo {...props} />,
  bank: (props) => <Bank {...props} />,
  bankInactive: (props) => <BankInactive {...props} />,
  statements: (props) => <Statements {...props} />,
  greenArrowDown: (props) => <GreenArrowDown {...props} />,
  debitCardBanner: (props) => <DebitCardBanner {...props} />,
  arrowRight: (props) => <ArrowRight {...props} />,
  arrowRightActive: (props) => <ArrowRightActive {...props} />,
  arrowRightInactive: (props) => <ArrowRightInactive {...props} />,
  activatedCard: (props) => <ActivatedCard {...props} />,
  changePin: (props) => <ChangePin {...props} />,
  findATMOption: (props) => <FindATMOption {...props} />,
  findDepositCash: (props) => <FindDepositCash {...props} />,
  lockCard: (props) => <LockCard {...props} />,
  replaceCard: (props) => <ReplaceCard {...props} />,
  accounts: (props) => <Accounts {...props} />,
  card: (props) => <Card {...props} />,
  debitCard: (props) => <DebitCard {...props} />,
  linkCard: (props) => <LinkCard {...props} />,
  accountsInactive: (props) => <AccountsInactive {...props} />,
  cardInactive: (props) => <CardInactive {...props} />,
  finEd: (props) => <FinEd {...props} />,
  finEdInactive: (props) => <FinEdInactive {...props} />,
  lightning: (props) => <Lightning {...props} />,
  arrowLeft: (props) => <ArrowLeft {...props} />,
  burgerMenu: (props) => <BurgerMenu {...props} />,
  errorIllustration: (props) => <ErrorIllustration {...props} />,
  successIllustration: (props) => <SuccessIllustration {...props} />,
  circledCheckmark: (props) => <CircledCheckmark {...props} />,
  menu: (props) => <Menu {...props} />,
  earlyPaycheck: (props) => <EarlyPaycheck {...props} />,
  logoSpecial: (props) => <LogoSpecial {...props} />,
  logoVertical: (props) => <LogoVertical {...props} />,
  overdraftProtection: (props) => <OverdraftProtection {...props} />,
  overdraftProtectionSettings: (props) => <OverdraftProtectionSettings {...props} />,
  directDeposit: (props) => <DirectDeposit {...props} />,
  backIcon: (props) => <BackIcon {...props} />,
  bankBlack: (props) => <BankBlack {...props} />,
  credit: (props) => <Credit {...props} />,
  debit: (props) => <Debit {...props} />,
  checkEmail: (props) => <CheckEmail {...props} />,
  emailCircled: (props) => <EmailCircled {...props} />,
  homeCircled: (props) => <HomeCircled {...props} />,
  home: (props) => <Home {...props} />,
  homeInactive: (props) => <HomeInactive {...props} />,
  userCircled: (props) => <UserCircled {...props} />,
  phoneCircled: (props) => <PhoneCircled {...props} />,
  termsSettings: (props) => <TermsSettings {...props} />,
  faqSettings: (props) => <FaqSettings {...props} />,
  directDepositSettings: (props) => <DirectDepositSettings {...props} />,
  accountInfoSettings: (props) => <AccountInfoSettings {...props} />,
  accountSummary: (props) => <AccountSummary {...props} />,
  aboutSettings: (props) => <AboutSettings {...props} />,
  contactSetting: (props) => <ContactSetting {...props} />,
  instantIcon: (props) => <InstantIcon {...props} />,
  giftOutlined: (props) => <GiftOutlined {...props} />,
  arrowsDown: (props) => <ArrowsDown {...props} />,
  close: (props) => <Close {...props} />,
  circledExclamation: (props) => <CircledExclamation {...props} />,
  setDebitCardPin: (props) => <SetDebitCardPin {...props} />,
  share: (props) => <Share {...props} />,
  showPassword: (props) => <ShowPassword {...props} />,
  hidePassword: (props) => <HidePassword {...props} />,
  contact: (props) => <Contact {...props} />,
  getPaidEarlyIllustration: (props) => <GetPaidEarlyIllustration {...props} />,
  bankCircle: (props) => <BankCircle {...props} />,
  depositCash: (props) => <DepositCash {...props} />,
  cardJumpingIllustration: (props) => <CardJumpingIllustration {...props} />,
  legalAgreementIllustration: (props) => <LegalAgreementIllustration {...props} />,
  finances: (props) => <Finances {...props} />,
  financesInactive: (props) => <FinancesInactive {...props} />,
  atm: (props) => <Atm {...props} />,
  checkmarkLines: (props) => <CheckmarkLines {...props} />,
  addBankAccount: (props) => <AddBankAccount {...props} />,
  activateCard: (props) => <ActivateCard {...props} />,
  parachuteCardIllustration: (props) => <ParachuteCard {...props} />,
  cardSuccessActivation: (props) => <CardSuccessActivation {...props} />,
  keyIcon: (props) => <Key {...props} />,
  closeSuccess: (props) => <CloseSuccess {...props} />,
  closeWhite: (props) => <CloseWhite {...props} />,
  closeAlert: (props) => <CloseAlert {...props} />,
  closeError: (props) => <CloseError {...props} />,
  closeInfo: (props) => <CloseInfo {...props} />,
  toastAlert: (props) => <ToastAlert {...props} />,
  toastInfo: (props) => <ToastInfo {...props} />,
  editIcon: (props) => <EditIcon {...props} />,
  playlist1: (props) => <Playlist1 {...props} />,
  reload: (props) => <Reload {...props} />,
  angleLeft: (props) => <AngleLeft {...props} />,
  angleRight: (props) => <AngleRight color="#fff" {...props} />,
};
