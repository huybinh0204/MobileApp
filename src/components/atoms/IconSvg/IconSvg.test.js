import { render } from '@testing-library/react-native';
import React from 'react';
import { ICONS } from '_constants';
import IconSvg from './IconSvg';

describe('IconSvg', () => {
  let props;

  beforeEach(() => {
    props = {
      icon: ICONS.logo,
      width: 100,
      height: 100,
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<IconSvg {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should Render mesage if the provided icon not exist', () => {
    const { getByTestId } = render(<IconSvg testID="icon" icon="testIcon" />);

    const icon = getByTestId('icon');

    expect(icon).toHaveTextContent('not Found');
  });
});
