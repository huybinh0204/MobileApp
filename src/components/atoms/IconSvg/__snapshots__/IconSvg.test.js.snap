// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSvg should match the snapshot 1`] = `
<SvgMock
  height={100}
  width={100}
/>
`;
